pkg/chartrenderer/default.go

```diff
@@ -159,42 +159,64 @@ func (c *RenderedChart) Manifest() []byte {
 	return b.Bytes()
 }
 
-// Files returns all rendered manifests mapping their names to their content.
-func (c *RenderedChart) Files() map[string]string {
-	var files = make(map[string]string)
+// Files returns a map representing the files associated with the RenderedChart.
+// The map has the file names as keys and their content represented as a nested map.
+// The nested map has keys in the format of strings.ToLower(manifest.Head.Kind+"/"+manifest.Head.Metadata.Name)
+// and values as the content of the corresponding file.
+func (c *RenderedChart) Files() map[string]map[string]string {
+	var files = make(map[string]map[string]string)
 	for _, manifest := range c.Manifests {
-		files[manifest.Name] = manifest.Content
+		if _, ok := files[manifest.Name]; ok {
+			files[manifest.Name][strings.ToLower(manifest.Head.Kind+"/"+manifest.Head.Metadata.Name)] = manifest.Content
+			continue
+		}
+		files[manifest.Name] = map[string]string{strings.ToLower(manifest.Head.Kind + "/" + manifest.Head.Metadata.Name): manifest.Content}
 	}
 	return files
 }
 
 // FileContent returns explicitly the content of the provided <filename>.
 func (c *RenderedChart) FileContent(filename string) string {
+	var fileContent strings.Builder
 	for _, mf := range c.Manifests {
 		if mf.Name == fmt.Sprintf("%s/templates/%s", c.ChartName, filename) {
-			return mf.Content
+			if fileContent.String() != "" {
+				// add `---` to seperate charts
```

Comment:
Nit: We are rather separate different manifest resources, not charts.

```suggestion
				// Add "---" to separate different resources
```